core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java

```diff
@@ -458,6 +458,78 @@ public void dayOfWeek() {
     assertEquals(integerValue(1), eval(expression));
   }
 
+  public void testDayOfWeekWithUnderscoress(FunctionExpression dateExpression, int dayOfWeek) {
+    assertEquals(INTEGER, dateExpression.type());
+    assertEquals(integerValue(dayOfWeek), eval(dateExpression));
+  }
+
+  @Test
+  public void dayOfWeekWithUnderscores() {
+    lenient().when(nullRef.valueOf(env)).thenReturn(nullValue());
+    lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
+
+    FunctionExpression expression1 = DSL.day_of_week(DSL.literal(new ExprDateValue("2020-08-07")));
+    FunctionExpression expression2 = DSL.day_of_week(DSL.literal(new ExprDateValue("2020-08-09")));
+    FunctionExpression expression3 = DSL.day_of_week(DSL.literal("2020-08-09"));
+    FunctionExpression expression4 = DSL.day_of_week(DSL.literal("2020-08-09 01:02:03"));
+
+    assertAll(
+        () -> testDayOfWeekWithUnderscoress(expression1, 6),
+        () -> assertEquals("day_of_week(DATE '2020-08-07')", expression1.toString()),
+
+        () -> testDayOfWeekWithUnderscoress(expression2, 1),
+        () -> assertEquals("day_of_week(DATE '2020-08-09')", expression2.toString()),
+
+
+        () -> testDayOfWeekWithUnderscoress(expression3, 1),
+        () -> assertEquals("day_of_week(\"2020-08-09\")", expression3.toString()),
+
+        () -> testDayOfWeekWithUnderscoress(expression4, 1),
+        () -> assertEquals("day_of_week(\"2020-08-09 01:02:03\")", expression4.toString())
+    );
+  }
+
+  public void testInvalidDayOfWeek(String date) {
```

Comment:
private

Reply:
Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/190/commits/f0448301a63c71a2f94f34fefc9e4840961cc437